### ISSUE TITLE
Make use of dynamic filters easier by adding website safety checker

### ISSUE THREAD

**Issue by mikhaelkh** [gorhill/uBlock#3596]
### Describe the issue
One way to quickly determine the broken filter is to add dynamic allow rules until site is rendered properly. Unfortunately, uBO (and uMatrix) provide no means to check if an unknown domain is safe. It can be done for example by adding a link to a website safety checker in the overview panel next to each domain.

### Steps for anyone to reproduce the issue
1. Enable `Advanced mode`
2. Go to website where exists broken filter
3. Try to fix it with dynamic filters

### Your settings

- OS/version: Windows 10 v1607
- Browser/version: Chrome Beta Official
- uBlock Origin version: Dev in Store

**Comment by gorhill** [comment]
uBO won't vouch for any "website safety checker".

**Comment by mikhaelkh** [comment]
You don't have to vouch for any particular extension if you give user a choice.
@jspenguin2017 ?

**Comment by jspenguin2017** [comment]
What?

**Comment by jspenguin2017** [comment]
> if you give user a choice

Like direct them to WebStore to pick their own? What's on your mind?

**Comment by gorhill** [comment]
Duplicate of #185 by the way.

**Comment by mikhaelkh** [comment]
> What's on your mind?

@jspenguin2017 more like use API to quickly see if the domain is safe, like red-yellow-green colors in the overview panel to see what requests from domain are currently blocked.
Maybe it makes sense to use separate color (say black) to indicate that there's an active rule which blocks whole domain? Currently green and red can turn yellow, but black will stay black.

**Comment by jspenguin2017** [comment]
> if you give user a choice

> use API to quickly see if the domain is safe

You just contradicted yourself.

**Comment by mikhaelkh** [comment]
Actually no, because there can be several APIs from several sources. Nevermind, it's like gorhill said, best leave it to a dedicated extension.

**Comment by jspenguin2017** [comment]
> from several sources

For example... ?

**Comment by mikhaelkh** [comment]
[WOT](https://www.mywot.com/wiki/API)
[Google](https://developers.google.com/safe-browsing/v4/)
[Yandex](https://tech.yandex.com/safebrowsing/)
[PhishTank ](https://www.phishtank.com/api_info.php)

**Comment by jspenguin2017** [comment]
WOT: 

> You need an API key to use the public API. In order to request a key, you need a WOT account.

Well, you can't really ship private API key with the extension, but the alternative is to host your own server, which isn't ideal. 

---

When Safe Browsing flags a site, it looks like this: 
![image](https://user-images.githubusercontent.com/7283682/37316709-44bdd3bc-2626-11e8-9e02-6c70f5d91654.png)

---

Yandex: 
> Get a unique key.

Well, same problem as WOT. 

---

PhishTank
> please register for an application key 

Well...

**Comment by mikhaelkh** [comment]
@jspenguin2017 
I guess my issue is that you can't distinguish dangerous sites from others in the overview panel. Another layer of protection like Google or Yandex Safe Browsing, or uMatrix is a smart choice in case you accidentally allow dangerous domain.

Maybe it makes sense to use separate color (say black) to indicate that there's an active block filter for domain from filterlist that contains unsafe websites, e.g. "Malware domains"? Currently green and red can turn yellow, but black will stay black.

**Comment by jspenguin2017** [comment]
@mikhaelkh 
That makes no sense to me. If a big full screen warning is not enough, how will a color code in a small grid help?

**Comment by mikhaelkh** [comment]
It's not "not enough", it's "too much". It helps that it would clearly indicate that the domain is unsafe with high probability and save time to find which domain to allow: first consider yellow domains, then red, and only then, with extreme care, black.

**Comment by jspenguin2017** [comment]
@mikhaelkh I don't think it's too much, there is no reason for you to visit phishing sites to begin with.

**Comment by mikhaelkh** [comment]
As I said, sometimes websites become victims of 3rd-party crap which gets blocked by GSB anyway.

**Comment by jspenguin2017** [comment]
@mikhaelkh Well, that's why you don't put untrusted third party stuff on your website.
